base/doc/ltnews38.tex

```diff
@@ -180,44 +310,49 @@ \section{Extended cross-referencing of document properties}
 \NewProperty{chapter}{now}{?}{\thechapter}
 \end{verbatim}
 The second argument means that the property is recorded immediately 
-(\enquote{now}) and not at shipout. The third argument sets a default if a 
+(\enquote{now}) and not at \cs{shipout}. The third argument sets a default if a 
 label is unknown. It is then possible to record a list of properties with 
 \cs{RecordProperties}: 
 \begin{verbatim}
-\RecordProperties{mylabel}{chapter,page,currentlabel}
+\RecordProperties{mylabel}
+                 {chapter,page,currentlabel}
 \end{verbatim}
 To reference the value the \cs{RefProperty} command is provided which takes 
 two arguments, the label and the property: 
 \begin{verbatim}
 \RefProperty{mylabel}{chapter}
 \end{verbatim}
  
-The new module predeclares a set of generally useful properties: 
-\texttt{currentlabel} (the standard counter representation), \texttt{page} 
-(the page representation), \texttt{title} (the title if set e.g.~by 
-\pkg{nameref}), \texttt{target} (the name of the link target if set e.g.~by 
-\pkg{hyperref}), \texttt{pagenum} (the page as an arabic number), 
-\texttt{abspage} (the absolute page number), \texttt{counter} (the name of 
-the current counter), \texttt{xpos} and \texttt{ypos} the position on the 
-page as set by a previous \cs{pdfsavepos}. 
+The new module predeclares a set of generally useful properties:
+\texttt{label} (the standard counter representation), \texttt{page}
+(the page representation), \texttt{title} (the title if set e.g.~by
+\pkg{nameref}), \texttt{target} (the name of the link target if set,
+e.g., by \pkg{hyperref}), \texttt{pagetarget} (the name of target
+added each page by \pkg{hyperref}), \texttt{pagenum} (the page as an
+arabic number), \texttt{abspage} (the absolute page number),
+\texttt{counter} (the name of the current counter), \texttt{xpos} and
+\texttt{ypos} the position on the page as set by a previous
+\cs{pdfsavepos}.
```

Comment:
“a previous” should probably be “the most recent (within the current shipout)”

Reply:
Done

Reply:
The effect of `\pdfsavepos` won't be reset during page shipout, hence the words "within the current shipout" (already added by commit 0eddf63d (Minor adjustment from Chris, 2023-10-26)) seem wrong.

Or maybe use "within the current compilation"?

```tex
% run with pdflatex-dev
\documentclass{article}
\begin{document}
first page
\ExplSyntaxOn
\tex_savepos:D
\newpage

last~ pos:~ \the\tex_lastxpos:D,~ \the\tex_lastypos:D
% typesets "last pos: 12664326, 46515936"
\ExplSyntaxOff
\end{document}
```

A lt-propertiles example
```tex
% run with pdflatex-dev
\documentclass{article}
\begin{document}
first page
\UseName{tex_savepos:D}

mypos is
at position (\RefProperty{mypos}{xpos} sp, \RefProperty{mypos}{ypos} sp)
on page~\RefProperty{mypos}{page}
% typesets "mypos is at position (12664326 sp, 46515936 sp) on page 2"
\newpage

second page
\RecordProperties{mypos}{xpos, ypos, page}
\end{document}
```

Reply:
It seems that I had misinterpreted what the PDFTeX manual said about its relationship to shipout. 

This example shows two things:
that the wording needs to be changed;
that such usage (not using or recording the saved information immediately) can easily result in the information being incorrect. 

Extra note: 
this engine-level behaviour (of savepos not being reset for each shipout) is perhaps questionable, and needs further investigation;
maybe this behaviour needs some mitigation at the LaTeX to avoid the recorded information being incomplete or misleading?

Reply:
I think it also means that LaTeX should explicitly reset thexpos/ypos property after shipout to 0/0